### ISSUE TITLE
buildEngineWithConfig(network, config); failure of TensorRT 8.2.4.2 when running CPP wts2trt on GPU 3060

### ISSUE THREAD

**Issue by wafaer** [NVIDIA/TensorRT#3487]
## Environment

**TensorRT Version**:8.2.4.2

**NVIDIA GPU**:3060

**NVIDIA Driver Version**:525

**CUDA Version**:11.4

**CUDNN Version**:8.2.4.15

Operating System:

Python Version (if applicable):

Tensorflow Version (if applicable):

PyTorch Version (if applicable):

Baremetal or Container (if so, version):

## Relevant Files

Hi, I'm trying to convert wts to float16 trtengine via CPP API, but it fails in buildEngineWithConfig(); Signal: SIGSEGV (Segmentation fault).

 The code for my network model was modified based on [here](https://github.com/wang-xinyu/tensorrtx/blob/master/yolov5/src/model.cpp), I replaced the backbone network c3 with c3attention, the C3attention model code is as follows:
static ILayer* convBlock1(INetworkDefinition *network, std::map<std::string, Weights>& weightMap, ITensor& input,
                          int outch, int ksize, int s, int g, std::string lname) {
    Weights emptywts{ DataType::kFLOAT, nullptr, 0 };
    int p = 3;
    IConvolutionLayer* conv1 = network->addConvolutionNd(input, outch, DimsHW{ ksize, ksize }, weightMap[lname + ".conv.weight"], emptywts);
    assert(conv1);
    conv1->setStrideNd(DimsHW{ s, s });
    conv1->setPaddingNd(DimsHW{ p, p });
    conv1->setNbGroups(g);
    conv1->setName((lname + ".conv").c_str());
    IScaleLayer* bn1 = addBatchNorm2d(network, weightMap, *conv1->getOutput(0), lname + ".bn", 1e-3);
    auto sig = network->addActivation(*bn1->getOutput(0), ActivationType::kSIGMOID);
    assert(sig);
    auto ew = network->addElementWise(*bn1->getOutput(0), *sig->getOutput(0), ElementWiseOperation::kPROD);
    assert(ew);
    return ew;
}
static ILayer* SpatialGate(INetworkDefinition *network, std::map<std::string, Weights>& weightMap, ITensor *input, std::string lname)
{
    Permutation permutation1 = {1,0,2};
    nvinfer1::IShuffleLayer* shuffleLayer1 = network->addShuffle(*input);
    shuffleLayer1->setFirstTranspose(Permutation(permutation1));
    ITensor* switch1 = shuffleLayer1->getOutput(0);
    Dims dims1 = switch1->getDimensions();
    int h_size = dims1.d[1];
    auto poolLayer = network->addPoolingNd(*switch1, PoolingType::kMAX,DimsHW{1,1});
    poolLayer->setStrideNd(DimsHW{1,1});
    poolLayer->setPaddingNd(DimsHW{0,0});
    poolLayer->setWindowSizeNd(DimsHW{h_size,1});
    ITensor* maxvalue = poolLayer->getOutput(0);
    Permutation permutation2 = {1,0,2};
    nvinfer1::IShuffleLayer* shuffleLayer2 = network->addShuffle(*maxvalue);
    shuffleLayer2->setFirstTranspose(Permutation(permutation2));
    ITensor* switch2 = shuffleLayer2->getOutput(0);
    Permutation permutation3 = {1,0,2};
    nvinfer1::IShuffleLayer* shuffleLayer3 = network->addShuffle(*input);
    shuffleLayer3->setFirstTranspose(Permutation(permutation3));
    ITensor* switch3 = shuffleLayer3->getOutput(0);
    Dims dims2 = switch3->getDimensions();
    int h_size1 = dims2.d[1];
    auto meanLayer = network->addPoolingNd(*switch3, PoolingType::kAVERAGE, DimsHW{1,1});
    meanLayer->setStrideNd(DimsHW{1,1});
    meanLayer->setPaddingNd(DimsHW{0,0});
    meanLayer->setWindowSizeNd(DimsHW{h_size1,1});
    ITensor* meanvalue = meanLayer->getOutput(0);
    Permutation permutation4 = {1,0,2};
    nvinfer1::IShuffleLayer* shuffleLayer4 = network->addShuffle(*meanvalue);
    shuffleLayer4->setFirstTranspose(Permutation(permutation4));
    ITensor* switch4 = shuffleLayer4->getOutput(0);
    ITensor* concatInputs[] = {switch2, switch4};
    ILayer* concatLayer = network->addConcatenation(concatInputs, 2);
    ITensor* input3 = concatLayer->getOutput(0);
    ILayer* convLayer = convBlock1(network,weightMap,*input3,1,7,1,1,lname + "spatial");
    ITensor* input4 = convLayer->getOutput(0);
    auto ew = network->addElementWise(*input, *input4, ElementWiseOperation::kPROD);
    return ew;
}

static ILayer* TripletAttention(INetworkDefinition *network, std::map<std::string, Weights>& weightMap, ITensor& input,std::string lname)
{
    Permutation permutation = {1,0,2};
    nvinfer1::IShuffleLayer* shuffleLayer = network->addShuffle(input);
    shuffleLayer->setFirstTranspose(Permutation(permutation));
    ITensor* perm1 = shuffleLayer->getOutput(0);
    ILayer* out1 = SpatialGate(network,weightMap,perm1,lname + ".ChannelGateH.");
    ITensor* out1_1 = out1->getOutput(0);
    Permutation permutation1 = {1,0,2};
    nvinfer1::IShuffleLayer* shuffleLayer1 = network->addShuffle(*out1_1);
    shuffleLayer1->setFirstTranspose(Permutation(permutation1));
    ITensor* out11 = shuffleLayer1->getOutput(0);
    Permutation permutation2 = {2,1,0};
    nvinfer1::IShuffleLayer* shuffleLayer2 = network->addShuffle(input);
    shuffleLayer2->setFirstTranspose(Permutation(permutation2));
    ITensor* perm2 = shuffleLayer2->getOutput(0);
    ILayer* out2 = SpatialGate(network,weightMap,perm2,lname + ".ChannelGateW.");
    ITensor* out2_1 = out2->getOutput(0);
    Permutation permutation3 = {2,1,0};
    nvinfer1::IShuffleLayer* shuffleLayer3 = network->addShuffle(*out2_1);
    shuffleLayer3->setFirstTranspose(Permutation(permutation3));
    ITensor* out21 = shuffleLayer3->getOutput(0);
    ILayer* x_out = SpatialGate(network,weightMap,&input,lname + ".SpatialGate.");
    ITensor* x_out1 = x_out->getOutput(0);
    ITensor* sum_input_tensors[] = {x_out1, out11, out21};
    ILayer* x1 = network->addElementWise(*sum_input_tensors[0],*sum_input_tensors[1], ElementWiseOperation::kSUM);
    ILayer* x2 = network->addElementWise(*(x1->getOutput(0)), *sum_input_tensors[2], ElementWiseOperation::kSUM);
    Dims dims = x2->getOutput(0)->getDimensions();
    const float num_tensors = 3.0;
    Weights scalarWeights{DataType::kFLOAT, &num_tensors, dims.d[0] * dims.d[1] * dims.d[2]};
    Dims scalarDims{dims.nbDims, {dims.d[0], dims.d[1], dims.d[2]}};
    ILayer* scalarTensor = network->addConstant(scalarDims, scalarWeights);
    ILayer* averageLayer = network->addElementWise(*(x2->getOutput(0)), *(scalarTensor->getOutput(0)), ElementWiseOperation::kDIV);
    return averageLayer;
}
static ILayer* C3attention(INetworkDefinition *network, std::map<std::string, Weights>& weightMap, ITensor& input, int c1, int c2, int n, bool shortcut, int g, float e, std::string lname) {
    int c_ = (int)((float)c2 * e);
    auto cv1 = convBlock(network, weightMap, input, c_, 1, 1, 1, lname + ".cv1");
    auto cv2 = convBlock(network, weightMap, input, c_, 1, 1, 1, lname + ".cv2");
    ITensor *y1 = cv1->getOutput(0);
    auto b = bottleneck(network, weightMap, *y1, c_, c_, shortcut, g, 1.0, lname + ".m.0");
    y1 = b->getOutput(0);
    ITensor* inputTensors[] = { y1, cv2->getOutput(0) };
    auto cat = network->addConcatenation(inputTensors, 2);
    ITensor* a3_input = cat->getOutput(0);
    auto a3 = TripletAttention(network,weightMap,*a3_input, lname + ".m1.0");
    a3_input = a3->getOutput(0);
    auto cv3 = convBlock(network, weightMap, *a3_input, c2, 1, 1, 1, lname + ".cv3");
    return cv3;
}

![Screenshot from 2023-11-24 17-20-16](https://github.com/NVIDIA/TensorRT/assets/92249545/ce09f1c0-8ea8-4cee-9cee-4a301cc73f58)
![Screenshot from 2023-11-24 17-20-26](https://github.com/NVIDIA/TensorRT/assets/92249545/eef7ddc6-7fd7-40a1-ad86-2fc458d400c7)

**[Model link](https://github.com/wang-xinyu/tensorrtx/blob/master/yolov5/src/model.cpp)**:

## Steps To Reproduce

**Commands or scripts**:

**Have you tried [the latest release](https://developer.nvidia.com/tensorrt)?**:

**Can this model run on other frameworks?** For example run ONNX model with ONNXRuntime (`polygraphy run <model.onnx> --onnxrt`):

**Comment by zerollzeng** [comment]
1. Does the original code work?
2. Can you try debug it use gdb? Segfault means memory error.

**Comment by wafaer** [comment]
> 1. Does the original code work?
> 2. Can you try debug it use gdb? Segfault means memory error.

1. The original code is working.
2. When I debug, ICudaEngine* engine = builder->buildEngineWithConfig(*network, *config); appears: Thread 1 "builder" received signal SIGSEGV, Segmentation fault.
0x00007fffdf0a9eeb in ?? ()  from /home/wg/library/TensorRT-8.2.4.2/lib/libnvinfer.so.8

**Comment by zerollzeng** [comment]
Segfault means memory error, make sure all the memory reference is created and the index didn't out of bound.

> The code for my network model was modified based on [here](https://github.com/wang-xinyu/tensorrtx/blob/master/yolov5/src/model.cpp), I replaced the backbone network c3 with c3attention, the C3attention model code is as follows:

You mentions you modify the code, does the reference weights still fit for the new code?

**Comment by wafaer** [comment]
> Segfault means memory error, make sure all the memory reference is created and the index didn't out of bound.
> 
> > The code for my network model was modified based on [here](https://github.com/wang-xinyu/tensorrtx/blob/master/yolov5/src/model.cpp), I replaced the backbone network c3 with c3attention, the C3attention model code is as follows:
> 
> You mentions you modify the code, does the reference weights still fit for the new code?

The weight file is trained after modifying the python version of the code.
